### PR TITLE
publish missing targets

### PR DESCRIPTION
**Pull request by i-walker**

### CODE REVIEW

**Comment by nomisRev** [comment]
Can we combine all files into a single file? That way it runs as a single Github action, meaning the jobs can short-circuit each other if they fail.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,71 +1,387 @@
...
+        working-directory: arrow-libs
+        run: ./gradlew publishMingwX64PublicationToDeployRepository
+  publish-arrow-stack:
     env:
       BASEDIR: ${{github.workspace}}/arrow-libs
-      JAVA_OPTS: -Xms512m -Xmx1024m
       SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
       SONATYPE_PWD: ${{ secrets.SONATYPE_PWD }}
       ORG_GRADLE_PROJECT_signingKey: ${{ secrets.ORG_GRADLE_PROJECT_signingKey }}
       ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.ORG_GRADLE_PROJECT_signingPassword }}
```
Comment (nomisRev) [inline .github/workflows/publish.yml:199]: I don't think we need to repeat this here if we have it at the very top.

**Review by nomisRev — APPROVED** [review]
LGTM. We should see SNAPSHOT appear for all binaries now.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -1,71 +1,387 @@
...
+      - main
+      - release/**
     paths:
-    - 'arrow-libs/**'
-    - '.github/**'
+      - 'arrow-libs/**'
+      - '.github/**'
+
+env:
+  BASEDIR: ${{github.workspace}}/arrow-libs
```
Comment (nomisRev) [inline .github/workflows/publish.yml:13]: This is not an env property shared for all jobs, is it?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -1,71 +1,387 @@
...
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Publish MingwX64
+        working-directory: arrow-libs
+        run: ./gradlew publishMingwX64PublicationToDeployRepository
+  publish-arrow-stack:
     env:
       BASEDIR: ${{github.workspace}}/arrow-libs
```
Comment (nomisRev) [inline .github/workflows/publish.yml:195]: Is this used? We changed to `working-directory: arrow-stack` in the `run` step.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -1,71 +1,387 @@
...
+          github-token: ${{github.token}}
+          script: |
+            await github.issues.create({...context.repo,
+             title: "[PUBLISH STACK] Error when merging '${{ steps.create-issue.outputs.title }}' on main branch",
+             body: "Please, check https://github.com/arrow-kt/arrow/actions/runs/${{ github.run_id }}"});
+
+  publish_doc:
+
+    env:
+      BASEDIR: ${{github.workspace}}/arrow-libs
```
Comment (nomisRev) [inline .github/workflows/publish.yml:244]: Think this is also unused here.

**Review by nomisRev — APPROVED** [review]
Looks great, I think there is a couple of redundant configurations.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (i-walker) [reply .github/workflows/publish.yml:13]: Yes it's used as an env variable for a script

File: .github/workflows/publish.yml (hunk 3 above)
Comment (i-walker) [reply .github/workflows/publish.yml:195]: the env variable is not affecting this step `working-directory: arrow-stack`

**Review by franciscodr — APPROVED** [review]
Thanks, @nomisRev!

**Comment by franciscodr** [comment]
And thanks to @i-walker, who has created this pull request 😂

**Pull request merged**